common/tools/dev-tool/src/templates/sampleReadme.md.ts

```diff
@@ -154,16 +157,47 @@ function createReadmeLink(info: SampleReadmeConfiguration) {
   return `https://github.com/Azure/azure-sdk-for-js/tree/main/${fragment}/README.md`;
 }
 
+/**
+ * Checks whether a url exists
+ * @param url -
+ * @returns true if the url is accessible; false otherwise
+ */
+async function urlExists(url: string): Promise<boolean> {
+  const res = await fetch(url);
+  return res.ok;
+}
+
+/**
+ * Creates link to the SDK package's Api reference page on learn.microsoft.com.
+ * @param info - the README configuration
+ * @returns a link to the api reference page.
+            This methods validates the generated api link first and will return an empty
+            string if the generated link is not valid.
+            No validation is done for customized api ref link that is passed in.
+ */
+async function createApiRef(info: SampleReadmeConfiguration): Promise<string> {
+  if (info.apiRefLink) {
+    return `[apiref]: ${info.apiRefLink}`;
+  } else if (info.scope.startsWith("@azure")) {
+    const link = `https://learn.microsoft.com/javascript/api/${info.scope}/${info.baseName}${info.isBeta ? "?view=azure-node-preview" : ""}`;
+    if (await urlExists(link)) {
+      return `[apiref]: ${link}`;
+    }
+    log.warn(`failed to reach api reference ${link} so not adding it.`);
+  }
+  return "";
+}
+
 /**
  * Creates a README for a sample package from a SampleReadmeConfiguration.
  */
-export default (info: SampleReadmeConfiguration): Promise<string> => {
+export default async (info: SampleReadmeConfiguration): Promise<string> => {
```

Comment:
Not sure why yet but making this async seems to cause dev-tool to hang

Reply:
could it be fetch() taking longer time?

Reply:
Likely this https://github.com/nodejs/node/issues/52909. Consuming body looks fixing it.  Thanks for the suggestion @xirzec!